src/tamp_improv/benchmarks/blocks2d_env.py

```diff
@@ -52,29 +90,16 @@ def is_block_in_target_area(
 
 
 class Blocks2DEnv(gym.Env):
-    """A block environment in 2D.
-
-    Observations are 15D:
-    - 4D for the x, y position (center), the width, and the height of
-    the robot
-    - 2D for the x, y position (center) of block 1 (the target block)
-    - 2D for the x, y position (center) of block 2 (the other block)
-    - 2D for the width and the height of the blocks
-    - 1D for the gripper "activation"
-    - 4D for the x, y position (center), the width, and the height of
-    the target area
-
-    Actions are 3D:
-    - 2D for dx, dy for the robot
-    - 1D for activating / deactivating the gripper
-
-    The environment has boundaries x=0 to x=1 and y=0 to y=1.
-    """
+    """A block environment in 2D with graph observation support."""
 
     metadata = {"render_modes": ["rgb_array"], "render_fps": 4}
 
-    def __init__(self, render_mode: str | None = None) -> None:
-        self.observation_space = Box(low=0, high=1, shape=(15,), dtype=np.float32)
+    def __init__(self, n_blocks: int = 2, render_mode: str | None = None) -> None:
+        self.n_blocks = min(max(2, n_blocks), 3)  # enforce limit between 2-3
```

Comment:
why? anyway, we should raise an error instead of min/maxing if the user specifies a different number of blocks, otherwise we might be unwittingly using a different number of blocks from what we expect